src/openMVG/multiview/solver_resection_p3p_nordberg.cpp

```diff
@@ -0,0 +1,289 @@
+        if (l1 >= 0.0)
+        {
+          Ls[valid] = Vec3(l1, l2, l3);
+          ++valid;
+        }
+      }
+    }
+  }
+
+  { //+v
+    double s = -v;
+    double w2 = 1.0 / (s * V(0, 1) - V(0, 0));
+    double w0 = (V(1, 0) - s * V(1, 1)) * w2;
+    double w1 = (V(2, 0) - s * V(2, 1)) * w2;
+
+    double a = 1.0 / ((a13 - a12) * w1 * w1 - a12 * b13 * w1 - a12);
+    double b = (a13 * b12 * w1 - a12 * b13 * w0 - 2.0 * w0 * w1 * (a12 - a13)) * a;
+    double c = ((a13 - a12) * w0 * w0 + a13 * b12 * w0 + a13) * a;
+
+    if (b * b - 4.0 * c >= 0)
+    {
+      double tau1, tau2;
+
+      root2real(b, c, tau1, tau2);
+      if (tau1 > 0)
+      {
+        double tau = tau1;
+        double d = a23 / (tau * (b23 + tau) + 1.0);
+        double l2 = std::sqrt(d);
+
+        double l3 = tau * l2;
+
+        double l1 = w0 * l2 + w1 * l3;
+        if (l1 >= 0)
+        {
+          Ls[valid] = Vec3(l1, l2, l3);
+          ++valid;
+        }
+      }
+      if (tau2 > 0)
+      {
+        double tau = tau2;
+        double d = a23 / (tau * (b23 + tau) + 1.0);
+        double l2 = std::sqrt(d);
+
+        double l3 = tau * l2;
+
+        double l1 = w0 * l2 + w1 * l3;
+        if (l1 >= 0)
+        {
+          Ls[valid] = Vec3(l1, l2, l3);
+          ++valid;
+        }
+      }
+    }
+  }
+
+  // if constexpr (refinement_iterations>0)
+  for (int i = 0; i < valid; ++i)
+  {
+    gauss_newton_refineL(Ls[i], a12, a13, a23, b12, b13, b23);
+  }
+
+  Vec3 ry1, ry2, ry3;
+  Vec3 yd1;
+  Vec3 yd2;
+  Vec3 yd1xd2;
+  Mat3 Xmat;
+  Xmat << d12(0), d13(0), d12xd13(0),
+          d12(1), d13(1), d12xd13(1),
+          d12(2), d13(2), d12xd13(2);
+  
+  Xmat = Xmat.inverse().eval();
+
+  for (int i = 0; i < valid; ++i)
+  {
+    // compute the rotation:
+    ry1 = f1 * Ls[i](0);
+    ry2 = f2 * Ls[i](1);
+    ry3 = f3 * Ls[i](2);
+
+    yd1 = ry1 - ry2;
+    yd2 = ry1 - ry3;
+    yd1xd2 = yd1.cross(yd2);
+
+    Mat3 Ymat;
+    Ymat << yd1(0), yd2(0), yd1xd2(0),
+            yd1(1), yd2(1), yd1xd2(1),
+            yd1(2), yd2(2), yd1xd2(2);
+
+    Mat3 Rs = Ymat * Xmat;
+    rotation_translation_solutions.emplace_back(Rs, ry1 - Rs * P1);
+  }
+  return valid;
+}
+
+void P3PSolver_Nordberg::Solve(
+    const Mat &bearing_vectors,
+    const Mat &X, // 3D points
+    std::vector<Mat34> *models)
+{
+  assert(3 == bearing_vectors.rows());
+  assert(3 == X.rows());
+  assert(bearing_vectors.cols() == X.cols());
+  Mat34 P;
+  std::vector<std::tuple<Mat3, Vec3>> rotation_translation_solutions;
+  if (computePosesNordberg(bearing_vectors, X, rotation_translation_solutions))
+  {
+    for (const auto & rot_trans_it : rotation_translation_solutions) {
+      Mat34 P;
+      P_From_KRt(Mat3::Identity(),           // intrinsics
+                  std::get<0>(rot_trans_it), // rotation
+                  std::get<1>(rot_trans_it), // translation
+                  &P);
+      models->push_back(P);
+    }
+  }
+  Mat solutions = Mat(3, 4 * 4);
```

Comment:
delete this line.

Reply:
ouch an useless alloc! nice catch! thanks!